projects/core/src/lib/components/toolbar/geolocation/geolocation.component.ts

```diff
@@ -0,0 +1,110 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, OnDestroy, OnInit } from '@angular/core';
+import { MapService, WKTHelper } from '@tailormap-viewer/map';
+import { Subject, takeUntil, take } from 'rxjs';
+import { Store } from '@ngrx/store';
+import { FeatureModel } from '@tailormap-viewer/api';
+
+@Component({
+  selector: 'tm-geolocation',
+  templateUrl: './geolocation.component.html',
+  styleUrls: ['./geolocation.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class GeolocationComponent implements OnInit, OnDestroy {
```

Comment:
Add some unit tests, to check if the expected methods are called when clicking the button.

Reply:
Added a single unit test to see if `watchLocation` is properly called, and if the map gets zoomed to properly.